apis/config/v1beta1/configuration_types.go

```diff
@@ -400,3 +404,15 @@ type FairSharing struct {
 	// The default strategy is ["LessThanOrEqualToFinalShare", "LessThanInitialShare"].
 	PreemptionStrategies []PreemptionStrategy `json:"preemptionStrategies,omitempty"`
 }
+
+type ObjectRetentionPolicies struct {
+	// FinishedWorkloadRetention is the duration to retain finished Workloads.
+	// A duration of 0 will delete finished Workloads immediately.
+	// A nil value will disable automatic deletion.
+	// The value is represented using the metav1.Duration format, allowing for flexible
+	// specification of time units (e.g., "24h", "1h30m", "30s").
+	//
+	// Defaults to null.
+	// +optional
+	FinishedWorkloadRetention *metav1.Duration `json:"finishedWorkloadRetention"`
```

Comment:
This seems like a relatively small and self-explanatory API, so I think we could go without KEP. WDYT @tenzen-y ?

Reply:
Yuki, do you have some history of adding small APIs without a KEP or extension of existing one?

On the second thought I think the new API can grow with new fields and having some place for discussing the new extensions, before implementing, would be useful. 

For that reason it could be helpful to already start with a KEP https://github.com/kubernetes/enhancements/blob/master/keps%2FREADME.md . Michal would you mind adding one based on the Kueue's template?

Reply:
> Yuki, do you have some history of adding small APIs without a KEP or extension of existing one?

Actually, we added the `active` field to the Workload without any KEP: https://github.com/kubernetes-sigs/kueue/issues/1091

But, as you know, the deactivation mechanism brings us implied specifications and bugs around the release 0.7, and we struggled to control and fix the bugs and implied specifications.
So, I regret that we did not prepare KEP for the deactivated mechanism.

Hence, I would like to prepare the dedicated KEP for this feature so that we can evaluate if this approach is better way.
@mimowo WDYT?

Reply:
Yeah, the KEP could be relatively short, but it would serve also as a basis for future extensions of the retention API if needed.

Reply:
@mimowo @tenzen-y Sure, I'm happy to prepare a KEP for it. I'm new to the project and learning its ways, so I'm sorry I didn't start with the KEP in the first place 🙇